lec05-dplyr.Rmd

```diff
@@ -1,4 +1,510 @@
+to solve scientific questions and to reproduce figures from the literature. To 
+answer solve these challenges, you will need to understand what data underlies a
+figure and how you need to manipulate it to recreate the figure.
+
+
+### Setting up
+
+Start by loading the required packages. Both **`ggplot2`** and **`dplyr`** are
+included in the **`tidyverse`** package collection.
+
+```{r}
+# Install if needed
+# install.packages('tidyverse')
+library(tidyverse)
+```
+
+### 1. Explore the weight and hindfoot trends further
+
+Load the data we saved in the previous lesson.
+
+```{r}
+# Download if needed
+# download.file("https://ndownloader.figshare.com/files/2292169", "data/portal_data.csv")
+surveys <- read_csv('portal_data.csv')
+surveys
+```
+
+Let's recreate the survey data set we used in lecture 4, only containing the most abundant species (those with > 800 observations).
+
+```{r}
+abundant_species <- surveys %>%
+    filter(!is.na(hindfoot_length) & !is.na(weight)) %>%
+    group_by(species) %>%
+    tally() %>%
+    arrange(desc(n)) %>%
+    filter(n > 800) %>%
+    select(species)
+
+surveys_abun_species <- surveys %>%
+    filter(!is.na(hindfoot_length) &
+        !is.na(weight) &
+        species %in% abundant_species$species)
+
+
+# If everything loaded correctly, the dimensions of your data set should be
+# 30,320 rows x 13 columns. We could check this by displaying the entire data
+# set with `surveys_abun_species`, but also more directly by comparing the
+# dimensions of the data set to those listed above.
+dim(surveys_abun_species) == c(30320, 13)
+# Remember that `==` is for comparisons, and `=` is for assigning arguments in
+# function calls.
+
+```
+
+In the second question in the last challenge of lecture 4, we saw that the
+average weight of all the animals decreased over time, while the average weight
+for each species remained constant. Here are those plots again.
+
+```{r}
+surveys_abun_species %>%
+    filter(!is.na(weight)) %>%
+    group_by(year) %>%
+    summarize(mean_weight = mean(weight)) %>%
+    ggplot(aes(x = year, y = mean_weight)) +
+        geom_line()
+```
+
+```{r}
+surveys_abun_species %>%
+    filter(!is.na(weight)) %>%
+    group_by(year, species) %>%
+    summarize(mean_weight = mean(weight)) %>%
+    ggplot(aes(x = year, y = mean_weight, color = species)) +
+        geom_line() +
+        facet_wrap(~ species)
+```
+
+If you were to look at the average hindfoot length over time, you would find 
+that the trends are similar to those of the average weight. Can you find an 
+explanation for why both the average hindfoot length and the average weight 
+decrease over time for all animals' average weight, but remain constant when 
+looking at individual species? Phrased another way: since each species display
+constant weight and hindfoot_length measures over time, what could be the cause
+to the notable decrease over time for the average weight of all species pooled
+together?
```

Comment:
This is actually known as the [Simpson's Paradox](https://en.wikipedia.org/wiki/Simpson%27s_paradox)! Super cool to see in this data!

Reply:
Cool!! So essentially varying sample sizes as a confounding variable? I didn't know that had a name!